runtime/builtin_types.go

```diff
@@ -552,6 +553,37 @@ func builtinUniChr(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	return NewUnicodeFromRunes([]rune{rune(i)}).ToObject(), nil
 }
 
+func builtinZip(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	seqs := [][]*Object{}
+	lenHint := math.MaxInt64
+	for _, arg := range args {
+		if raised := checkFunctionArgs(f, "zip", Args{arg}, ObjectType); raised != nil {
+			return nil, raised
+		}
+		seq, raised := seqNew(f, Args{arg})
```

Comment:
It's expensive to allocate all these slices, so I would recommend creating an iterator for each arg and calling next over all the iterators until one is exhausted.

That means you won't get a lenHint which makes it a little more expensive to build the result list, but you can grow the list slice by doubling its capacity every time to minimize the number of reallocations.